llvm/test/Transforms/InstCombine/xor.ll

```diff
@@ -1502,8 +1502,8 @@ entry:
 define <2 x i32> @or_disjoint_with_xor_vec(<2 x i32> %a, < 2 x i32> %b, <2 x i32> %c) {
 ; CHECK-LABEL: @or_disjoint_with_xor_vec(
 ; CHECK-NEXT:  entry:
-; CHECK-NEXT:    [[OR:%.*]] = or disjoint <2 x i32> [[A:%.*]], [[B:%.*]]
-; CHECK-NEXT:    [[XOR:%.*]] = xor <2 x i32> [[OR]], [[C:%.*]]
+; CHECK-NEXT:    [[TMP0:%.*]] = xor <2 x i32> [[A:%.*]], [[C:%.*]]
+; CHECK-NEXT:    [[XOR:%.*]] = xor <2 x i32> [[TMP0]], [[B:%.*]]
 ; CHECK-NEXT:    ret <2 x i32> [[XOR]]
 ;
 entry:
```

Comment:
Can you please add more negative test cases? 
you are using `m_OneUse` we should have a test to check if this works as expected.
and a test for the `C ^ (A | B)` case.

Reply:
Done